src/Generic/Data/Internal/Enum.hs

```diff
@@ -63,6 +84,27 @@ class GEnum opts f where
 -- report](https://www.haskell.org/onlinereport/haskell2010/haskellch11.html#x18-18400011.2)).
 data StandardEnum
 
+-- | Extends the 'StandardEnum' option for 'GEnum' to allow all constructors to 
+-- have arbitrary many arguments. Each argument type/field must be an instance 
+-- of both 'Enum' and 'Bounded'. Two restrictions require the user's caution:
+--
+-- * The instances of the field types need to be valid. Particularly 'Int' is an 
+-- unfit field type, because the enumeration of the negative values starts 
+-- before 0. 
+--
+-- * The generic type must not exceed the enumeration limit. As 'Enum' converts 
+-- from and to 'Int', at most @(maxBound :: Int)@ + 1 many values can be
+--  enumerated. This restriction makes 'Word' an invalid field type. Notably it
+-- is insufficient for each individual field types to stay below the limit.
```

Comment:
Since the cardinality is explicitly represented as an Int there can only be up to maxBound values.